sql-plugin/src/main/scala/com/nvidia/spark/rapids/collectionOperations.scala

```diff
@@ -16,10 +16,75 @@
 
 package com.nvidia.spark.rapids
 
-import ai.rapids.cudf.ColumnVector
+import scala.collection.mutable.ArrayBuffer
+
+import ai.rapids.cudf.{ColumnVector, ColumnView}
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
 
 import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.unsafe.types.UTF8String
+
+case class GpuConcat(children: Seq[Expression]) extends GpuComplexTypeMergingExpression {
+
+  @transient override lazy val dataType: DataType = {
+    if (children.isEmpty) {
+      StringType
+    } else {
+      super.dataType
+    }
+  }
+
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  override def columnarEval(batch: ColumnarBatch): Any = dataType match {
+    case StringType => stringConcat(batch)
+    case ArrayType(_, _) => listConcat(batch)
+  }
+
+  private def stringConcat(batch: ColumnarBatch): GpuColumnVector = {
+    val rows = batch.numRows()
+
+    withResource(ArrayBuffer.empty[ColumnVector]) { buffer =>
+      withResource(GpuScalar.from(null, StringType)) { nullScalar =>
```

Comment:
Nit: We know the exact type we need, so no need to dispatch on the type at runtime.
```suggestion
      withResource(Scalar.fromString(null)) { nullScalar =>
```